client/src/NoteEdit.js

```diff
@@ -1,19 +1,60 @@
-class NoteEdit extends React.Component {
+import React, { Component } from 'react';
+import $ from 'jquery';
+import debounce from 'throttle-debounce/debounce';
+import bootbox from 'bootbox';
+import './keyboard';
+import NoteList from './NoteList';
+import NoteForm from './NoteForm';
+import Note from './Note';
+import PushState from  './PushState';
+import Navbar from './Navbar';
+import SaveStateLabel from './SaveStateLabel';
+import AddNoteButton from './AddNoteButton';
+import AlertFlash from './AlertFlash';
+import EventHive from './EventHive';
+import AutoSave from './AutoSave';
+
+class NoteEdit extends Component {
   constructor(props, context) {
     super(props, context);
 
+    this.pushState = new PushState(this.props.history);
+
+    const { match } = this.props;
+
     if (props.note) {
       this.state = {
         isInitialEdit: true,
         note: Note.fromAttributes(props.note)
       };
-      PushState.setBrowserTitle(props.note);
+      this.pushState.setBrowserTitle(props.note);
+    }
+    else if (match.params.id) {
+      // set new note as state, otherwise the state will be undefined
+      this.state = Object.assign(this.getNewNoteAttributes(), { isInitialEdit: true });
```

Comment:
depending on your transpiler/babel-config you can use object-spread:

```
this.state = {
   isInitialEdit: true,
   ...this.getNewNoteAttributes()
}

```

this makes it more readable, in particular, because you see that `this.getNewNoteAttributes()` would overwrite `isInitialEdit`, if it would return an object with that property.